drivers/storage/portworx/component/prometheus.go

```diff
@@ -0,0 +1,514 @@
+	ownerRef *metav1.OwnerReference,
+) error {
+	return k8sutil.CreateOrUpdateClusterRoleBinding(
+		c.k8sClient,
+		&rbacv1.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:            PrometheusClusterRoleBindingName,
+				OwnerReferences: []metav1.OwnerReference{*ownerRef},
+			},
+			Subjects: []rbacv1.Subject{
+				{
+					Kind:      "ServiceAccount",
+					Name:      PrometheusServiceAccountName,
+					Namespace: clusterNamespace,
+				},
+			},
+			RoleRef: rbacv1.RoleRef{
+				Kind:     "ClusterRole",
+				Name:     PrometheusClusterRoleName,
+				APIGroup: "rbac.authorization.k8s.io",
+			},
+		},
+		ownerRef,
+	)
+}
+
+func (c *prometheus) createOperatorDeployment(
+	cluster *corev1alpha1.StorageCluster,
+	ownerRef *metav1.OwnerReference,
+) error {
+	existingDeployment := &appsv1.Deployment{}
+	err := c.k8sClient.Get(
+		context.TODO(),
+		types.NamespacedName{
+			Name:      PrometheusOperatorDeploymentName,
+			Namespace: cluster.Namespace,
+		},
+		existingDeployment,
+	)
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+
+	if !c.isOperatorCreated || errors.IsNotFound(err) {
+		deployment := getPrometheusOperatorDeploymentSpec(cluster, ownerRef)
+		if err = k8sutil.CreateOrUpdateDeployment(c.k8sClient, deployment, ownerRef); err != nil {
+			return err
+		}
+	}
+	c.isOperatorCreated = true
+	return nil
+}
+
+func getPrometheusOperatorDeploymentSpec(
+	cluster *corev1alpha1.StorageCluster,
+	ownerRef *metav1.OwnerReference,
+) *appsv1.Deployment {
+	replicas := int32(1)
+	runAsNonRoot := true
+	runAsUser := int64(65534)
+	labels := map[string]string{
+		"k8s-app": PrometheusOperatorDeploymentName,
+	}
+	operatorImage := util.GetImageURN(cluster.Spec.CustomImageRegistry, "quay.io/coreos/prometheus-operator:v0.29.0")
+	args := make([]string, 0)
+	args = append(args,
+		fmt.Sprintf("--kubelet-service=%s/kubelet", cluster.Namespace),
+		"--config-reloader-image=quay.io/coreos/configmap-reload:v0.0.1",
+	)
+
+	return &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            PrometheusOperatorDeploymentName,
+			Namespace:       cluster.Namespace,
+			OwnerReferences: []metav1.OwnerReference{*ownerRef},
+			Labels:          labels,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Replicas: &replicas,
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: v1.PodSpec{
+					ServiceAccountName: PrometheusOperatorServiceAccountName,
+					Containers: []v1.Container{
+						{
+							Name:  "px-prometheus-operator",
+							Image: operatorImage,
+							Args:  args,
+							Ports: []v1.ContainerPort{
+								{
+									Name:          "http",
+									ContainerPort: int32(8080),
+								},
+							},
+							Resources: v1.ResourceRequirements{
+								Limits: map[v1.ResourceName]resource.Quantity{
+									v1.ResourceCPU:    resource.MustParse("200m"),
```

Comment:
What happens if someone externally updates these resource requirements on the deployment? Will the operator reconcile it back to these values?

Reply:
It does an Update so it will be overwritten again.

Reply:
As discussed offline, this will get overwritten as we don't store the deployment config. For now removing the resource completely and letting k8s allocate resources to the pod. If there is a need we can expose an annotation to override the cpu/mem like we do for pvc-controller, stork and autopilot deployments.